### DISCUSSION TITLE
Welcome to Computational Tools for Research Discussions!

### DISCUSSION THREAD

**Discussion by magitz** [comptoolsres/comptoolsres.github.io#5] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other students and the instructor. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

## Class Demeanor and Netiquette
The discussion boards can be used to ask for and provide help by all. Students should be supportive and considerate of others at all times. Rude or inappropriate comments will be removed and the poster will be warned.

## Getting started
To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by plehman2000** [comment]
Hello, my name is Patrick Lehman. I'm a computer engineering undergrad student with an interest in doing neuroscience research. I'm hoping this class will equip me the tools to analyze biological data

**Reply by admontalvo** [reply]
Hello Everyone,

My name is Abelardo D. Montalvo. I am a second year PhD student in the College of Pharmacy in the Department of Pharmacotherapy and Translational Research. I recently started working with large genomic datasets involving single nucleotide polymorphisms. I plan to use this course to further learn about the command line and various ways to organize and display data as well as the integration with Python.

**Reply by admontalvo** [reply]
I have been going through the Github exercises and whenever I use the command git push I end up getting the following message after putting my GitHub log in information: 

remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/"username"/hello_world.git/'

Has anybody been able to resolve this issue?
It appears that I need to create a token or some sort of way to authenticate my GitHub account but am not entirely sure.

**Reply by magitz** [reply]
Hi @Morivivi, Thanks for posting on the discussion board!
This is a bit of a pain and we will (hopefully) cover it in class today.
We will setup ssh keys for `git push/pull` on HiPerGator.

**Reply by magitz** [reply]
I have updated the github page on the class website with this information--thanks for reminding me that I needed to add that.
[https://comptoolsres.github.io/TLCL_4.html](https://comptoolsres.github.io/TLCL_4.html)

**Reply by admontalvo** [reply]
I am not sure if anybody else has been having similar issues, but I have decided to put the following comments in this discussion board.

In the Gainesville_weather_1960-2022_fixed.csv the AWND column displays the dates for some reason. This is regarding question 4 of problem set 1.

For question 5, I ended up using the old file for (Gainesville_weather_1960-2022.csv) for the grep question rather than the fixed weather file and was able to find the date with no problem. 

When I tried to use the fixed weather file, I got a message that Gainesville_weather_1960-2022_fixed.csv no such file or directory found. 
I used grep "month/day/year" and got the above message.

Additionally for question 6 on problem set 1, are we supposed to individually separate each column using regular expressions and treat them as a variable and then replacement with the desired output or can we just target all the columns and replace it with the desired output? Additionally, would we have to target the subheadings or also take into consideration the values below them?  

I am probably overthinking this, but I am not sure if anybody has thought about this. Any feedback from anyone is greatly appreciated.

**Reply by magitz** [reply]
Hi @admontalvo 
For the first issue, remember that the Name column still has "GAINESVILLE REGIONAL AIRPORT, FL US", so if you use commas as the delimiter, that will count as 2 columns and the headers will be off by one after that.

Fine to use either the older file or the "fixed" one.

For question 5, the file is there...
```
[magitz@login5 data]$ pwd
/blue/bsc4452/share/Class_Files/data
[magitz@login5 data]$ head Gainesville_weather_1960-2022_fixed.csv 
STATION,NAME,DATE,AWND,PRCP,SNOW,SNWD,TAVG,TMAX,TMIN,WDF5,WSF5
USW00012816,"GAINESVILLE REGIONAL AIRPORT, FL US",5/1/1960,,,0,0,,,,,
USW00012816,"GAINESVILLE REGIONAL AIRPORT, FL US",5/2/1960,,,0,0,,,,,
USW00012816,"GAINESVILLE REGIONAL AIRPORT, FL US",5/3/1960,,,0,0,,,,,
USW00012816,"GAINESVILLE REGIONAL AIRPORT, FL US",5/4/1960,,,0,0,,,,,
USW00012816,"GAINESVILLE REGIONAL AIRPORT, FL US",5/5/1960,,,0,0,,,,,
USW00012816,"GAINESVILLE REGIONAL AIRPORT, FL US",5/6/1960,,,0,0,,,,,
USW00012816,"GAINESVILLE REGIONAL AIRPORT, FL US",5/7/1960,,,0,0,,,,,
USW00012816,"GAINESVILLE REGIONAL AIRPORT, FL US",5/8/1960,,,0,0,,,,,
USW00012816,"GAINESVILLE REGIONAL AIRPORT, FL US",5/9/1960,,,0,0,,,,,
```
For the last one, there are many ways to di it, my strategy is typically write 1 regular expression that matches each column for a line, add parentheses around the parts for the columns you want to keep and then use those captured parts in the replacement using the $1, $2, etc variables from the capture.

Hope that helps.

**Comment by admontalvo** [comment]
I am not sure if anybody else has received this message but whenever, I try to use git push to move a script to github after creating a branch, cloning the repository, and committing it I get the following message:

ERROR: Permission to comptoolsres/Class_Files.git denied to admontalvo.
fatal: Could not read from remote repository.

I verified that my ssh key is in my github account and that the repository that we created as a team was successfully cloned.

Any advice is greatly appreciated.
